dataactbroker/app.py

```
@@ -15,97 +12,79 @@
-        else:
-            cors = CORS(app, supports_credentials=True, origins=CONFIG_SERVICES['cross_origin_url'],
-                        allow_headers = "*", expose_headers = "X-Session-Id")
-        # Enable AWS Sessions
-        app.session_interface = DynamoInterface()
-        # Set up bcrypt
-        bcrypt = Bcrypt(app)
-
-        @app.teardown_appcontext
-        def teardown_appcontext(exception):
-            GlobalDB.close()
-
-        @app.before_request
-        def before_request():
-            GlobalDB.db()
-
-        # Root will point to index.html
-        @app.route("/", methods=["GET"])
-        def root():
-            return "Broker is running"
-
-        if local:
-            localFiles = os.path.join(broker_file_path, "<path:filename>")
-            # Only define this route when running locally
-            @app.route(localFiles)
-            def sendFile(filename):
-                if(config["local"]) :
-                    return send_from_directory(broker_file_path, filename)
-        else:
-            # For non-local installs, set Dynamo Region
-            SessionTable.DYNAMO_REGION = CONFIG_BROKER['aws_region']
-
-        # Add routes for modules here
-        add_login_routes(app, bcrypt)
-
-        add_file_routes(app, CONFIG_BROKER['aws_create_temp_credentials'],
-            local, broker_file_path, bcrypt)
-        add_user_routes(app, app.config['SYSTEM_EMAIL'], bcrypt)
-        add_domain_routes(app, local, bcrypt)
-
-        SessionTable.LOCAL_PORT = CONFIG_DB['dynamo_port']
-
-        SessionTable.setup(app, local)
-
-        if local:
-            checkDynamo()
-
-        return app
-
-    except Exception as e:
-        exc_type, exc_obj, exc_tb = sys.exc_info()
-        trace = traceback.extract_tb(exc_tb, 10)
-        CloudLogger.logError('Broker App Level Error: ', e, trace)
-
-        del exc_tb
-        raise
```

If I understand correctly, @jworcestBAH is correct -- application initialization failures would be captured in uwsgi log rather than CloudLogger. That seems like an okay thing, however, given that the exception would be raised when the application was _initializing_. uwsgi logs should include the traceback, yes (it'd be on separate lines, however). In general, I'm a big proponent of ripping out CloudLogger wherever possible and preferring the more standard (and configurable) `logging` system, or (as in this case), deferring to the underlying runner (uwsgi, celery, etc.) for log capture.